src/main/scala/util/Hierarchy.scala

```diff
@@ -0,0 +1,110 @@
+// See LICENSE for license details.
+
+package sifive.blocks.util
+
+import chisel3._
+import chisel3.util._
+
+import freechips.rocketchip.config._
+import freechips.rocketchip.diplomaticobjectmodel.logicaltree.LogicalTreeNode
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.prci._
+import freechips.rocketchip.util.LocationMap
+
+import firrtl.graph._
+
+case object HierarchyKey extends Field[Option[DiGraph[HierarchicalLocation]]](None)
+
+case class DevicesSubsystemParams(
+  name: String,
+  logicalTreeNode: LogicalTreeNode,
+  asyncClockGroupsNode: ClockGroupEphemeralNode)
+
```

Comment:
I think you should put these classes in Devices.scala since they don't really have that much to do with Hierarchy other than being its (current) target